linbox/blackbox/ntl-hankel.inl

```diff
@@ -117,12 +117,12 @@ namespace LinBox
 	       	Father_t(v.field())
 	{
 		//! @warning Assumes that the input is a vector of ZZ_p else things will FAIL
-		if ( isEven( v.size() ) )
+		if ( isEven(uint64_t( v.size() )) )
```

Comment:
There are a lot of `size_t` -> `uint64_t` in the PR, what's the rationale behind that?

If it is because of warnings, I feel like `isEven()` is missing an overload.

Reply:
+1: `size_t` should be used for indexes and sizes, while `int64_t` is for ring/field elements.

Reply:
In fact this is a problem introduced in the design of Givaro where all creation/conversion with integers must be of type (u)int[8,16,32,64]_t or float/double.
On OSX the type size_t is aliased to the type long long, and unfortunately none of the (u)int[8,16,32,64]_t correspond to long long. Therefore, using size_t to create or convert an integer will not compile at all on OSX.

There are two solutions:
- either we modify Givaro to allow size_t (and probably ptrdiff_t) as a possible type for integer
- or we use ad'hoc conversion to uint64_t when necessary: when converting to integer

Note that we have no other choice as the STL method size() from container return a size_t.
After some discussion with JGD and Clement it appears that ad'hoc hacking must be preferred.
But we need to be very attentive with OSX building on Jenkins.

Reply:
I totally agree.
As I said, I think we should let size_t deal with indices, and not numbers (ring/field elements). Hence, we should not have to call convert or init from a `size_t`.
As far as `isEven` is concerned, it seems that the only implem is over Givaro::Integer, hence your `size_t` is first cast to an `unt64_t` and then to a GMP  `mpz_t`.
I would much rather have a straight `(v.size() & size_t(1) ) or define the isEven specialization for `size_t` with above code in linbox.

Reply:
So these are just validated changes to compile on OSX. Ok for me.

Reply:
Opened #137.